src/rsysreqs/scripts/scripts.go

```diff
@@ -0,0 +1,21 @@
+package scripts
+
+import (
+	"errors"
+
+	"rsysreqs/rules"
+)
+
+var ErrUnsupportedSystem = errors.New("unsupported system")
+
+type ScriptGenerator interface {
+	InstallScripts(packages []string) []string
+}
+
+func NewScriptGenerator(sys rules.System) (ScriptGenerator, error) {
+	switch {
+	case sys.Os == "linux" && sys.Distribution == "ubuntu":
+		return ubuntuScriptGenerator{}, nil
+	}
+	return nil, ErrUnsupportedSystem
+}
```

Comment:
Is it odd that this package depends on the rules package just for the `System` struct? I thought it would be more convenient than recreating all the same args (os, dist, arch, etc.) which would need to be validated or converted to enums at some point.

Reply:
Depending on the scenario, I've sometimes created a separate package (usually called `metadata`) that I use simply for structs that are shared across multiple packages. This can save you some headache with import cycles if you need to use some structs in a lot of different packages. For now, though, I'd say just wait -- you can always move them later if it's necessary.